homeassistant/components/powerwall/switch.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Powerwall Switches (V2 API only)."""
+
+from typing import Any
+
+from tesla_powerwall import GridStatus, IslandMode, Powerwall
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, POWERWALL_API, POWERWALL_COORDINATOR
+from .models import PowerwallRuntimeData
+
+ICON = "mdi:home-battery-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Powerwall switch platform from Powerwall resources."""
+    powerwall_data: PowerwallRuntimeData = hass.data[DOMAIN][config_entry.entry_id]
+    power_wall = powerwall_data[POWERWALL_API]
+    coordinator = powerwall_data[POWERWALL_COORDINATOR]
+    assert coordinator is not None
+
+    async_add_entities([PowerwallOffGridEnabledEntity(power_wall, coordinator)])
+
+
+class PowerwallOffGridEnabledEntity(SwitchEntity):
```

Comment:
This inheritance looks like it's missing the base Powerwall entity

Reply:
There doesn't seem to be a base Powerwall entity in this integration?  Looking at some similar integrations, this would be a strongly typed version of the coordinator, this is already being passed in?

Looks like adding this would end up being a very large change unless I'm mistaken?

Reply:
It should be a small change: https://github.com/home-assistant/core/pull/86357/files#r1083649806